libpkg/pkgdb_iterator.c

```diff
@@ -1116,47 +1116,42 @@ int
 pkgdb_it_next(struct pkgdb_it *it, struct pkg **pkg_p, unsigned flags)
 {
 	struct pkg_repo_it *rit;
-	int ret;
+	int ret = EPKG_FATAL;
 
 	assert(it != NULL);
 
-	switch (it->type) {
-	case PKGDB_IT_LOCAL:
-		return (pkgdb_sqlite_it_next(&it->un.local, pkg_p, flags));
-		break;
-	case PKGDB_IT_REPO:
-		if (it->un.remote != NULL) {
-			rit = it->un.remote->it;
-			ret = rit->ops->next(rit, pkg_p, flags);
-			if (ret != EPKG_OK) {
-				/*
-				 * Detach this iterator from list and switch to another
-				 */
-				struct _pkg_repo_it_set *tmp;
-
-				rit->ops->free(rit);
-				tmp = it->un.remote;
-				it->un.remote = tmp->next;
-				free(tmp);
-
-				return (pkgdb_it_next(it, pkg_p, flags));
-			}
-
-			if (*pkg_p != NULL)
-				(*pkg_p)->repo = rit->repo;
+	if (it->local != NULL && !it->local->finished) {
+		if (pkgdb_sqlite_it_next(it->local, pkg_p, flags) == EPKG_OK)
+			return EPKG_OK;
```

Comment:
you can simplify by just directly using (return (pkgdb_sqlite_it_nex(...))

Reply:
I did that at first, but it didn't work. For some reason, `it->local->finished` is 0, but when the execution enters inside the if, `pkgdb_sqlite_it_next()` returns !EPKG_OK. I just thought I misunderstood the `finsihed` field. Is this a bug?